hypothesis-python/src/hypothesis/internal/conjecture/junkdrawer.py

```diff
@@ -268,8 +268,13 @@ def __underlying_index(self, i: int) -> int:
                 i += 1
         return i
 
+    # even though we have len + getitem, mypyc requires iter.
+    def __iter__(self) -> Iterator[T]:
+        for i in range(len(self)):
+            yield self[i]
 
-def swap(ls: LazySequenceCopy, i: int, j: int) -> None:
+
+def swap(ls: LazySequenceCopy[T], i: int, j: int) -> None:
     """Swap the elements ls[i], ls[j]."""
```

Comment:
Not about the type annotations, but this seems like it should be an instance method of `LazySequenceCopy` rather than a standalone function.